editor/plugins/control_editor_plugin.cpp

```diff
@@ -792,230 +771,300 @@ void ControlEditorToolbar::_selection_changed() {
+		} else {
+			container_h_picker->set_allowed_flags_all();
+			container_v_picker->set_allowed_flags_all();
+		}
+	} else {
+		CanvasItemEditor::get_singleton()->set_tool_drawer_visible(this, false);
+	}
+}
 
-	container_h_presets_menu = memnew(MenuButton);
-	container_h_presets_menu->set_shortcut_context(this);
-	container_h_presets_menu->set_text(TTR("Horizontal"));
-	add_child(container_h_presets_menu);
-	container_h_presets_menu->set_switch_on_hover(true);
+ContainerEditorToolbar::ContainerEditorToolbar(EditorNode *p_editor) {
+	EditorToolDrawerGroup *container_h_presets_group = memnew(EditorToolDrawerGroup);
+	container_h_presets_group->set_title(TTR("Horizontal alignment"));
+	add_child(container_h_presets_group);
 
-	p = container_h_presets_menu->get_popup();
-	p->connect("id_pressed", callable_mp(this, &ControlEditorToolbar::_popup_callback));
+	container_h_picker = memnew(ContainerSizeFlagPicker(false));
+	container_h_presets_group->add_child(container_h_picker);
+	container_h_picker->connect("size_flags_selected", callable_mp(this, &ContainerEditorToolbar::_container_flags_selected), varray(false));
 
-	container_v_presets_menu = memnew(MenuButton);
-	container_v_presets_menu->set_shortcut_context(this);
-	container_v_presets_menu->set_text(TTR("Vertical"));
-	add_child(container_v_presets_menu);
-	container_v_presets_menu->set_switch_on_hover(true);
+	EditorToolDrawerGroup *container_v_presets_group = memnew(EditorToolDrawerGroup);
+	container_v_presets_group->set_title(TTR("Vertical alignment"));
+	add_child(container_v_presets_group);
 
-	p = container_v_presets_menu->get_popup();
-	p->connect("id_pressed", callable_mp(this, &ControlEditorToolbar::_popup_callback));
+	container_v_picker = memnew(ContainerSizeFlagPicker(true));
+	container_v_presets_group->add_child(container_v_picker);
+	container_v_picker->connect("size_flags_selected", callable_mp(this, &ContainerEditorToolbar::_container_flags_selected), varray(true));
 
 	undo_redo = p_editor->get_undo_redo();
 	editor_selection = p_editor->get_editor_selection();
 	editor_selection->add_editor_plugin(this);
-	editor_selection->connect("selection_changed", callable_mp(this, &ControlEditorToolbar::_selection_changed));
-
-	singleton = this;
+	editor_selection->connect("selection_changed", callable_mp(this, &ContainerEditorToolbar::_selection_changed));
 }
 
-ControlEditorToolbar *ControlEditorToolbar::singleton = nullptr;
-
 ControlEditorPlugin::ControlEditorPlugin(EditorNode *p_editor) {
 	editor = p_editor;
 
-	toolbar = memnew(ControlEditorToolbar(editor));
-	toolbar->hide();
-	add_control_to_container(CONTAINER_CANVAS_EDITOR_MENU, toolbar);
+	anchors_toolbar = memnew(ControlAnchorsEditorToolbar(editor));
+	CanvasItemEditor::get_singleton()->add_control_to_tool_drawer(TTR("Control Anchors"), editor->get_gui_base()->get_theme_icon(SNAME("ControlLayout"), SNAME("EditorIcons")), anchors_toolbar);
+
+	container_toolbar = memnew(ContainerEditorToolbar(editor));
```

Comment:
You don't need to pass `editor` as parameter, because it's a singleton. See #57306